### PR TITLE
Dockerfile fixes

### PR DESCRIPTION
**Pull request by mpreisler**
Simplified and sped up the Dockerfile build. Removed some unnecessary stuff from `BUILD.md`.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 ## From source
 
-1. On Red Hat Enterprise Linux and Fedora make sure the packages `cmake`, `openscap-utils`, `openscap-python`, and `python-lxml` and their dependencies are installed. We require version `1.0.8` or later of `openscap-utils` (available in Red Hat Enterprise Linux) as well as `git`.
+1. On Red Hat Enterprise Linux and Fedora make sure the packages `cmake`, `openscap-utils` and and their dependencies are installed. We require version `1.0.8` or later of `openscap-utils` (available in Red Hat Enterprise Linux) as well as `git`.
 
- `# yum -y install cmake openscap-utils openscap-python python-lxml git`
+ `# yum -y install cmake openscap-utils git`
 
- On Ubuntu make sure the packages `expat`, `libopenscap8`, `libxml2-utils`, `python-lxml`, `python-openscap`, and `xsltproc` and their dependencies are installed as well as `git`.
+ On Ubuntu make sure the packages `expat`, `libopenscap8`, `libxml2-utils`, `xsltproc` and their dependencies are installed as well as `git`.
```
Comment (redhatrises) [inline BUILD.md]: **,** and their dependencies

File: BUILD.md [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 ## From source
 
-1. On Red Hat Enterprise Linux and Fedora make sure the packages `cmake`, `openscap-utils`, `openscap-python`, and `python-lxml` and their dependencies are installed. We require version `1.0.8` or later of `openscap-utils` (available in Red Hat Enterprise Linux) as well as `git`.
+1. On Red Hat Enterprise Linux and Fedora make sure the packages `cmake`, `openscap-utils` and and their dependencies are installed. We require version `1.0.8` or later of `openscap-utils` (available in Red Hat Enterprise Linux) as well as `git`.
```
Comment (redhatrises) [inline BUILD.md]: **,** and their dependencies

File: Dockerfile [hunk 3]
```diff
@@ -14,9 +15,11 @@ WORKDIR /home/$OSCAP_USERNAME
...
+# clean the build dir in case the user is also building SSG locally
+RUN rm -rf $OSCAP_DIR/build/*
+
 WORKDIR /home/$OSCAP_USERNAME/$OSCAP_DIR/build
 
 RUN cmake ..
 
-ENTRYPOINT ["/usr/bin/make"]
-CMD ["all"]
+CMD /usr/bin/make -j $BUILD_JOBS
```
Comment (redhatrises) [inline Dockerfile]: Shouldn't this be `CMD ["/usr/bin/make", "-j", "$BUILD_JOBS"]`?

Comment (konstruktoid) [inline Dockerfile]: Docker can't really handle environmental variables well @redhatrises, so this is an acceptable solution.

File: BUILD.md [hunk 4]
```diff
@@ -138,4 +138,4 @@ To build content only for a specific distribution, add the relevant name as a fl
 
 Using `docker cp` to copy all the generated content to the your host:
 
-`$ docker cp scap-security-guide:/home/oscap/scap-security-guide/build $(pwd)`
+`$ docker cp scap-security-guide:/home/oscap/scap-security-guide/build $(pwd)/container_build`
```
Comment (konstruktoid) [inline BUILD.md]: Just nitpicking, but is this necessary? The path will be `container_build/build` and the `container_build` dir has to exist.

Comment (mpreisler) [inline BUILD.md]: I do not want to suggest overwriting anything in the docs. Won't this just copy the container dir into `./container_build`? Similar to what `cp` would do?

File: Dockerfile (hunk 3 above)
Comment (mpreisler) [inline Dockerfile]: @redhatrises yeah, unfortunately, this is necessary if you want to use env vars in `CMD`.

File: BUILD.md (hunk 1 above)
Comment (mpreisler) [inline BUILD.md]: done

File: BUILD.md (hunk 2 above)
Comment (mpreisler) [inline BUILD.md]: done

File: BUILD.md (hunk 4 above)
Comment (mpreisler) [inline BUILD.md]: Confirmed with `man docker-cp`:

```
 · SRC_PATH specifies a directory

       · DEST_PATH does not exist

       · DEST_PATH is created as a directory and the contents of the source directory are copied into this directory
```

**Comment by redhatrises** [comment]
Ack.

**Pull request merged**